### DISCUSSION TITLE
Baresip as a WebRTC-SIP Gateway?

### DISCUSSION THREAD

**Discussion by joenye** [baresip/baresip#2424] [Q&A]
I have an Asterisk server that I cannot modify (vendor product) that exposes SIP over TCP/UDP on port 5060 (no TLS). I'm able to connect to it using Baresip and initiate/receive calls over RTP/RTSP with h264 video, PCMU/PCMA audio 👍 

However, I would like desktop and mobile browsers to initiate and receive calls to/from the Asterisk server over websocket and WebRTC completely transparently (without changing the Asterisk server).

Is this something I can do with Baresip? Is this something I can do by modfying the [webrtc demo](https://github.com/baresip/baresip/tree/main/webrtc)? Or is Baresip is designed for client devices only and I'd have to look elsewhere for this?

**Comment by nanosonde** [comment]
Maybe this project is interesting for you?
https://github.com/florian-h05/webrtc-sip-gw

Basically it is meant wo be used in a small private smart home setup.
It makes it possible to connect to any SIP server via websocket (SIP-over-websocket) instead of only UDP/TCP.
The project was specifically tailored to be used with a home router called Fritzbox which offers a small SIP PBX.

**Reply by joenye** [reply]
Thank you @nanosonde this looks really good. I had seen [havfo/WEBRTC-to-SIP](https://github.com/havfo/WEBRTC-to-SIP) before but it looked like it needed lots of revision to fit into my smart home  Docker-based setup.

So I'm very glad to see this project you've linked as it looks you've based it on this work and tidied things up. However I don't have a Fritzbox and don't intend to buy one either. Would you still suggest I investigate/modify your project in this case?

My use case (based on your GitHub repos may be similar to yours) is to connect my intercom to [TECH7Fox/sip-hass-card](https://github.com/TECH7Fox/sip-hass-card). I went down the baresip etc. route as it feels overkill running a full PBX like Asterisk. Janus Gateway looks like it won't work with JSSIP either.

**Reply by nanosonde** [reply]
Yes, the project is based on the mentioned project. 
However, I also stopped working on this and handed it over to @florian-h05.
Feel free to modify/extend the project for your own needs.
